pkg/kubelet/network/cni/cni.go

```diff
@@ -43,6 +44,8 @@ type cniNetworkPlugin struct {
 	host           network.Host
 	execer         utilexec.Interface
 	nsenterPath    string
+	mu             sync.Mutex
+	podMutexes     map[string]*sync.Mutex
```

Comment:
For the normal, graceful deletions, thel pod worker (goroutine) is responsible for killing the pod, and there should not be race conditions. What @dcbw talked about is the case where a pod gets deleted non-gracefully (i.e., `---grace-period=0`), and kubelet's global cleanup goroutine will try to kill the pod regardless of what the pod worker is doing. This race condition is the root cause and should be fixed: https://github.com/kubernetes/kubernetes/issues/13311#issuecomment-229214387